tests/test_thermostate.py

```diff
@@ -937,3 +937,41 @@ def test_set_hv(self):
         assert np.isclose(s.v, Q_(0.4772010021515822, "m**3/kg"))  # type: ignore
         assert np.isclose(s.h, Q_(1061602.391543017, "J/kg"))  # type: ignore
         assert np.isclose(s.x, Q_(0.28475636946248034, "dimensionless"))  # type: ignore
+
+        
+    def test_state_units_EE(self):
+        """ Set a state with EE units and check the properties."""
+        s = State("water", T = Q_(100, 'degC'), p = Q_(1.0, 'atm'), units = "EE")
+        assert s.units == "EE"
+        assert s.cv.units == "british_thermal_unit / degree_Rankine / pound"
+        assert s.cp.units == "british_thermal_unit / degree_Rankine / pound"
+        assert s.s.units == "british_thermal_unit / degree_Rankine / pound"
+        assert s.h.units == "british_thermal_unit / pound"
+        assert s.T.units == "degree_Fahrenheit"
+        assert s.u.units == "british_thermal_unit / pound"
+        assert s.v.units == "pound_force_per_square_inch"
+        assert s.p.units == "foot ** 3 / pound"
+        
+    def test_state_units_SI(self): 
+        """ Set a state with SI units and check the properties."""
+        s = State("water", T = Q_(100, 'degC'), p = Q_(1.0, 'atm'), units = "SI")
+        assert s.units == "SI"
+        assert s.cv.units == "kilojoule / kelvin / kilogram"
+        assert s.cp.units == "kilojoule / kelvin / kilogram"
+        assert s.s.units == "kilojoule / kelvin / kilogram"
+        assert s.h.units == "kilojoule / kilogram"
+        assert s.T.units == "degree_Celsius"
+        assert s.u.units == "kilojoule / kilogram"
+        assert s.v.units == "bar"
+        assert s.p.units == "meter ** 3 / kilogram"
+              
+    def test_default_units(self):
+        """ Set default units and check for functionality"""
+        s = State("water", T = Q_(100, 'degC'), p = Q_(1.0, 'atm'))
+        assert s.units is None
+        set_default_units("SI")
+        s2 = State("water", T = Q_(100, 'degC'), p = Q_(1.0, 'atm'))
+        assert s2.units == "SI"
+        set_default_units("EE")
```

Comment:
I think one of the reasons the tests are failing is because there's no guarantee of the order that the tests run. Since you're changing the default units here, and `default_units` is a module-level variable, it affects _all the `State`s created after this is set_! I don't think this will be a problem in general usage, since this would tend to get changed for a single problem, and then left alone. For the purposes of the test suite though, we need to figure out a way to "unset" this.